amiadapters/beacon.py

```diff
@@ -516,3 +528,81 @@ def _transformed_meter_output_file(self) -> str:
 
     def _transformed_reads_output_file(self) -> str:
         return os.path.join(self.output_folder, f"{self.name()}-transformed-reads.txt")
+
+
+class BeaconSnowflakeStorageSink(SnowflakeStorageSink):
```

Comment:
Relying heavily on a long-ish chain of inheritance for this, which isn't my favorite. Holler if you have other ideas!